docs(tuto): add github actions tutorial (#1370)
scaleway/docs-content: tutorials/use-container-registry-github-actions/index.mdx

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [created a private Container Registry namespace](/containers/container-registry/how-to/create-namespace)
  - You have [configured your API Keys](/identity-and-access-management/iam/how-to/create-api-keys/)
  - You have [the member role in a Github organization](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization)
  - You have [created a Github repository](https://docs.github.com/en/get-started/quickstart/create-a-repo) in that organization
  - You have [created a working Dockerfile in the root of that repository](https://docs.docker.com/engine/reference/builder/) in that organization
</Message>

## Setting up your Scaleway secrets in Github

In order for your Github Action to interact with your Scaleway Container Registry, it is necessary to provide the required data to your repository. The first step is to store all the necessary information in Github secrets.

> 1. Go to the **Settings** page of your repository. In the "Secrets and variables" menu, select **Actions**.

Reviewer: ```suggestion
1. Go to the **Settings** page of your Github repository. In the "Secrets and variables" menu, select **Actions**.
```

> 1. Go to the **Settings** page of your Github repository. In the "Secrets and variables" menu, select **Actions**.

Reviewer 2: ```suggestion
1. Go to the **Settings** page of your Github repository. In the **Secrets and variables** menu, select **Actions**.
```

> 2. We will create a first secret holding your Scaleway API key.

Reviewer: ```suggestion
2. Create a first secret holding your Scaleway API key.
```

    Click **New repository secret**. In the "name" field, enter `SCALEWAY_API_KEY`, and paste your API secret key in the "value" field. The Access Key ID will not be used here.
    <Message type="important">

>         Beware to not leave any leading or trailing spaces in the "value" field.

Reviewer: ```suggestion
        Make sure not to leave any leading or trailing spaces in the "value" field.
```

    </Message>

> 3. We will create a second secret holding your container registry endpoint. You can find your endpoint in your Scaleway Container Registry settings:

Reviewer: ```suggestion
3. Create a second secret holding your container registry endpoint. You can find your endpoint in your Scaleway Container Registry settings:
```

    <Lightbox src="scaleway-container-registry-settings.webp" alt="" />

    Click **New repository secret** again. In the "name" field, enter `CONTAINER_REGISTRY_ENDPOINT`, and paste your endpoint in the "value" field.

## Creating your Github Action workflow

> We will create a GitHub Actions workflow that you can trigger, which is an automated sequence of steps to build and deploy your Docker image.

Reviewer: ```suggestion
Next we create a GitHub Actions workflow that you can trigger, which is an automated sequence of steps to build and deploy your Docker image.
```

### Creating the workflow

> Github workflows are described by [YAML](https://en.wikipedia.org/wiki/YAML) files located in the `.github/workflows` directory from your repository.

Reviewer: ```suggestion
Github workflows are described by [YAML](https://en.wikipedia.org/wiki/YAML) files located in the `.github/workflows` directory of your repository.
```

In your repository root, create the `.github` folder and the `workflows` sub-folder, then create a file `.github/workflows/build.yml`.

> ```

Reviewer 2: ```suggestion
```yml
```

name: Build & Deploy to Scaleway

on:
  workflow_dispatch:

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - name: Login to Scaleway Container Registry
      uses: docker/login-action@v2
      with:
        username: nologin
        password: ${{ secrets.SCALEWAY_API_KEY }}
        registry: ${{ secrets.CONTAINER_REGISTRY_ENDPOINT }}
    - name: Build the Docker image
      run: docker build . -t ${{ secrets.CONTAINER_REGISTRY_ENDPOINT }}/my-docker-image
    - name: Push the Docker Image
      run: docker push ${{ secrets.CONTAINER_REGISTRY_ENDPOINT }}/my-docker-image
```

> This workflow has one job called "build", which is set to run on the latest version of Ubuntu. The job includes four steps:

Reviewer 2: ```suggestion
This workflow has one job called **build**, which is set to run on the latest version of Ubuntu. The job includes four steps:
```

1. `actions/checkout@v3` is used to check out the repository's code.
2. `docker/login-action@v2` is used to authenticate with the Scaleway Container Registry. The action uses the API key and container registry endpoint stored in Github secrets.
3. `docker build` is used to build a Docker image with the code from the repository. The resulting image is tagged with the container registry endpoint and the name of the image, which is `my-docker-image` in this case.
4. `docker push` is used to push the Docker image to the Scaleway Container Registry.

<Message type="note">
    The `on.workflow_dispatch` node in the workflow YAML tree specifies that this workflow can be run manually in Github, which is the path that we chose in this tutorial. There are [many other events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) that you can choose from to trigger your workflow, such as a push or a merge.
</Message>

### Launch the workflow

> The workflow is now ready to get deployed. Commit and push your `build.yml` file, then get back to your repository's home page on Github.

Reviewer: ```suggestion
The workflow is now ready to be deployed. Commit and push your `build.yml` file, then go back to your repository's home page on Github.
```
